src/infrastructure/database/models/user/user_database_handler.py

```diff
@@ -164,3 +170,22 @@ async def search_users_by_phrase(
             {'username': {'$regex': phrase, '$options': 'i', '$ne': current_user['username']}}).skip(offset).limit(
             limit))
         return result
+
+    @staticmethod
+    async def add_to_favourite_counter(
+            collection: Collection,
+            user_id: ObjectId
+    ):
+        user = collection.find_one({'_id': user_id})
+        favourites_count = user['favourites_count'] + 1
+        collection.update_one({'_id': {'$eq': ObjectId(user_id)}}, {'$set': {'favourites_count': favourites_count}})
+
+    @staticmethod
+    async def remove_from_favourite_counter(
+            collection: Collection,
+            user_id: ObjectId
+    ):
+        user = collection.find_one({'_id': user_id})
+        if user["favourites_count"] > 0:
+            favourites_count = user['favourites_count'] - 1
+            collection.update_one({'_id': {'$eq': ObjectId(user_id)}}, {'$set': {'favourites_count': favourites_count}})
```

Comment:
Może dałoby się zamiast 2 operacji zrobić jedną?
Połączyć $cond z $inc

Reply:
"The $cond is an aggregation operator you can not use in a simple update query,"
Znalezione na stacku - faktycznie sypało mi błędy :c

Reply:
to użyj collection.aggregate ?

Reply:
```
// { a: "Hello", b: "World" }
// { a: "Olleh", b: "Dlrow" }
db.collection.updateMany(
  {},
  [ { $set: { active: { $eq: [ "$a", "Hello" ] } } } ]
)
// { a: "Hello", b: "World", active: true  }
// { a: "Olleh", b: "Dlrow", active: false }

Reply:
A jakby zrobić krok wstecz

Reply:
```
collection.update_one({'_id': {'$eq': ObjectId(user_id)}, 'favourites_count': {'$gt': 0}}, {'$set': {'favourites_count': {'$inc': 1}}})
```
I zrobić coś takiego